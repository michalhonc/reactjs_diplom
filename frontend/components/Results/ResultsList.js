import React from 'react';
import {  } from 'prop-types';
import styled from 'styled-components';

const NON_EXTENDED_ITEM = 5;

const StyledListItem = styled.li`
    list-style: none;
    display: flex;
    justify-content: space-between;
    border-bottom: 2px solid #eee;
    padding: 0.5rem;
`;

const StyledList = styled.ul`
    padding: 0;
    margin: 0;
`;

const StyledButton = styled.button`
    border: 0;
    padding: 0;
    margin: 0;
    background-color: transparent;
    color: #999;
    font-size: 1.4rem;
    line-height: 2rem;
    text-decoration: underline;
    cursor: pointer;
`;

const ResultsList = ({ list }) => {
    const[isExtended, setIsExtended] = React.useState(false);
    const listToRender = isExtended ? list : list.slice(0, NON_EXTENDED_ITEM);
    return (
        <>
            <StyledList>
                {listToRender.map((item, index) => {
                    return (
                        <StyledListItem key={item.id}>
                            <span>{index + 1},</span>
                            <span>{item.full_name}</span>
                            <span>{item.size}tis.</span>
                        </StyledListItem>
                    )
                })}
            </StyledList>
            {!isExtended && (
                    <StyledButton onClick={() => setIsExtended(true)}>
                        Zobrazit další
                    </StyledButton>
                )
            }
        </> 
    )
};

//ResultsList.propTypes = {
//	
//}
//
//ResultsList.defaultProps = {
//	
//}

export default ResultsList;
