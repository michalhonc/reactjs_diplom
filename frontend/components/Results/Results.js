import React from 'react';
import styled from 'styled-components';
import { string } from 'prop-types';
import ResultsList from './ResultsList';
import useDebounce from '../../lib/useDebounce';

const url = 'https://api.github.com/users/';
const MAX_ITEM = 10;

const StyledWrapper = styled.div`
    margin: 2rem 3rem;
`;

const Results = (props) => {
    const [list, useList] = React.useState([]);
    const [isError, useIsError] = React.useState(false);
    const [isFetching, useIsFetching] = React.useState(false);
    const debouncedQuery = useDebounce(props.query, 500);

    const req = async () => {
        const fetchUrl = `${url}${debouncedQuery}/repos`;
        useIsFetching(true);
        useIsError(false);
        try {
            const response = await (await fetch(fetchUrl)).json();
            const out = response.slice(0, MAX_ITEM).sort((a, b) => b.size - a.size);
            useIsFetching(false);
            useList(out);
        } catch(e) {
			console.log('TCL: }catch -> e', e)
            useIsFetching(false);
            useIsError(true);
        }
    }

    React.useEffect(() => {
        req();
    }, [debouncedQuery])
    
    if (isError) {
        return <p>Jejda, spadlo to..<button onClick={req}>Zkusit znova</button></p>
    }

    if (isFetching) {
        return  <p>Načítám data..</p>
    }

    return (
        <StyledWrapper>
            {Array.isArray(list) && list.length > 0 && (
                <ResultsList list={list} />
            )}
        </StyledWrapper>
    );
};

Results.propTypes = {
    query: string,
}

Results.defaultProps = {
    query: 'gaearon'
}

export default Results;