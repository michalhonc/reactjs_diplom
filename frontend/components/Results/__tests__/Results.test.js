import React from 'react';
import Results from '../Results';
import renderer from 'react-test-renderer';

it('Results render 5 items', async () => {
    fetch.mockResponse(() => callMyApi().then(res => ({body: res})));

    const tree = renderer.create(<Results query="gaearon" />);
    await Promise.resolve();
    expect(tree.toJSON()).toMatchSnapshot();
});
