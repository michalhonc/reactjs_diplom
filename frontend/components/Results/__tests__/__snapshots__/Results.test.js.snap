// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results render 5 items 1`] = `
<div
  className="Results__StyledWrapper-sc-1g02b5t-0 DeKtb"
/>
`;
